Police-Data-Accessibility-Project/data-sources-app-v2
# Mc 443 locality table

### Fixes

* https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/443
* https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/450

### Description

* Updates app to account to database location/agencies overhaul, which normalizes location logic and simplifies agency access, while also making a number of QoL improvements to `agencies` and various locational tables.
* Schema logic has been extended for the `/agencies` endpoint, with enhanced schema validation and schema documentation.
* This is a considerable overhaul, and more of an overhaul than I'm typically comfortable making, but which to some degree felt unavoidable due to how interconnected the backend logic was. In the future, I'll be trying to keep changes more focused and less sprawling.

### Testing

* Run tests in `/tests` directory
* Confirm that `/agencies` endpoints continue to function as expected, and that documentation matches logic.

### Performance

* Schema and locational normalization logic, as well as other QoL improvements, add marginal performance overhead increases

### Docs

* API documentation updated 

### Breaking Changes

* The `/agencies` `POST` and `PUT` endpoints have been modified considerably from their prior form, and now accept their arguments in different JSON formats
* The `/agencies` `GET` endpoints have changed in what they've returned.
